conference.js

```diff
@@ -505,26 +528,21 @@ export default {
                     mediaPermissionPromptVisibilityChanged(true, browser))
         );
 
-        // First try to retrieve both audio and video.
         let tryCreateLocalTracks;
 
-        // FIXME the logic about trying to go audio only on error is duplicated
-        if (options.startAudioOnly) {
-            tryCreateLocalTracks
-                = createLocalTracks({ devices: ['audio'] }, true)
-                    .catch(err => {
-                        audioOnlyError = err;
-
-                        return [];
-                    });
+        // Enable audio only mode
+        if (config.startAudioOnly) {
+            APP.store.dispatch(toggleAudioOnly());
+        }
 
-            // Enable audio only mode
-            if (config.startAudioOnly) {
-                APP.store.dispatch(toggleAudioOnly());
-            }
-        } else if (options.startScreenSharing) {
+        // FIXME is there any simpler way to rewrite this spaghetti below ?
```

Comment:
Maybe for a start we can add it to the mediaDeviceHelper class or a similar one. At least it will be isolated.